### ISSUE TITLE
Remove explicit dependency on the Kotlin stdlib 

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#122]
More info in https://kotlinlang.org/docs/whatsnew14.html#dependency-on-the-standard-library-added-by-default

**Comment by DriblingTrex** [comment]
Please help me out I'm not getting what to do as this is my first try to open source

**Comment by vestrel00** [comment]
@DriblingTrex, welcome!!! 🔥 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥

Thanks for taking interest in helping me with this issue ❤️ 

I have assigned you this issue. It is yours 😁 

Given that you are new to open source, here is a guide on what you can do to complete and close this issue! 

## Creating your first Pull Request to Contacts, Reborn!

1. Download the latest version of Android Studio; https://developer.android.com/studio
2. Fork this repo.
    - <img width="747" alt="fork" src="https://user-images.githubusercontent.com/4332753/144131711-ec477dba-d896-462b-93ca-de58b3d6cb27.png">
3. Clone your fork to your local machine.
    - <img width="495" alt="clone" src="https://user-images.githubusercontent.com/4332753/144132041-e9e54b67-e096-4df5-a3d9-1467a8b0121b.png">
    - You will need to have installed git and know the basics of cloning. Here is the official guide for it https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository
4. Open up the project in Android Studio.
5. Build and run the sample app to make sure everything is working fine for you.
    - <img width="406" alt="run-sample" src="https://user-images.githubusercontent.com/4332753/144133179-a6db8857-8616-43e4-8188-d7fb42a1f070.png">
6. When you get the build working and the sample app running, it's time to make the code changes required to close this issue! You can take a look at the **Making the changes** section below for a guide on what to do.
7. Build and run the sample app to make sure everything is working fine for you.
    - <img width="406" alt="run-sample" src="https://user-images.githubusercontent.com/4332753/144133179-a6db8857-8616-43e4-8188-d7fb42a1f070.png">
8. Commit and push up your changes.
    - ```
      git add .
      git commit -m "Closes #122 Remove explicit dependency on the Kotlin stdlib"
      git push -u origin main
      ```
9. Create a new pull request from your fork to this repo.
    - <img width="660" alt="pr-1" src="https://user-images.githubusercontent.com/4332753/144134218-836c34b8-4eba-4b5a-8e11-9593546404b4.png">
    - <img width="759" alt="pr-2" src="https://user-images.githubusercontent.com/4332753/144134225-6861cc03-6ff6-4090-84d6-acf83fba559b.png">
10. And that's it! I will review your changes, approve it, and then merge it if everything looks good.

Once your pull request is merged into this repo, you will officially be considered a Contributor by GitHub 😀. Your commit(s) will be a part of the commit history of this repo. The commit will be linked to your GitHub account to make sure you get credit ❤️ 

## Making the changes

This project is currently using Kotlin version 1.4.21, which means that it actually does not need to explicitly declare the Kotlin `stdlib` in any of its modules as stated in https://kotlinlang.org/docs/whatsnew14.html#dependency-on-the-standard-library-added-by-default

<img width="894" alt="kotlin-update" src="https://user-images.githubusercontent.com/4332753/144135392-8953fafe-cc55-4ad4-8af1-d3a7e8371bd7.png">

This means that we can remove the following line of code in the root `build.gradle` file,

```groovy
buildscript {
  ext {
    ...
    libraries = [
      ...
      kotlin: [
        ...
        standardLib      : "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$versions.kotlin.core", // Remove this!
      ]
    ]
  }
  ...
}
```

Then, remove the following code in all of the submodule `build.gradle` files,

```groovy
dependencies {
    ...
    implementation libraries.kotlin.standardLib // Remove this!
}
```

Do this for all submodules; `async`, `core`, `customdata-gender`, `customdata-handlename`, `debug`, `permissions`, `sample`, `test`, and `ui`.

Once you've done all this, clean the project and run the sample app.
<img width="514" alt="clean" src="https://user-images.githubusercontent.com/4332753/144136286-5275795c-b5ca-4d1f-88e5-d264ae01762d.png">
<img width="406" alt="run-sample" src="https://user-images.githubusercontent.com/4332753/144136291-663d8bba-7b04-49f0-a650-5dd51e48641e.png">

If the build succeeds and the sample app works, you are done!

If you get compile errors for some reason, then just try running it again. If that still does not work, I'd suggest invalidating cache and restart.

<img width="439" alt="invalidate-restart" src="https://user-images.githubusercontent.com/4332753/144136509-44623fb1-c571-4e7f-93c5-21f8abed212a.png">

If that still does not work, file the PR anyways and I'll help you get it to work 😁 

GLHF!

**Comment by vestrel00** [comment]
@DriblingTrex, I noticed you've forked this repo already. Nice! Btw, I pushed up some changes in the main branch from #121 so you might want pull the latest changes into your fork if you still want to tackle this issue 😁 

There is no rush with this issue. It's not needed for anything so you can literally do it in a month or whenever you feel like it. Just let me know if you don't want to anymore. Thanks in advance ❤️

**Comment by DriblingTrex** [comment]
@vestrel00 thanks very much for your guidance . I am trying to configure all things and surely put my efforts to do this task.
At last thanking you for assigning me this task.

**Comment by DriblingTrex** [comment]
@vestrel00 I'm not able to push changed file, git is showing message "remote: vestrel00/contacts-android.git denied to DriblingTrex"
Help me out please!!

**Comment by vestrel00** [comment]
This is a part of [v0.1.9 Release Checklist](https://github.com/vestrel00/contacts-android/discussions/132).

**Comment by DriblingTrex** [comment]
Glad to know I contributed in project. please guide me to move forward from here.

**Comment by vestrel00** [comment]
@DriblingTrex, well, I'd like you to show up as a contributor so maybe you can try to fix your commits so that your GitHub account is linked to it? https://github.com/vestrel00/contacts-android/pull/133#issuecomment-988072393
